apps/omg_watcher/test/support/integration/test_helper.ex

```diff
@@ -41,6 +41,29 @@ defmodule OMG.Watcher.Integration.TestHelper do
     |> wait_for(timeout)
   end
 
+  @spec wait_for_byzantine_events_match([match_function :: (any() -> bool())], timeout()) :: {:ok, any()} | no_return()
```

Comment:
this helper is complex enough to `@doc` it

**EDIT** on second thought this helper looks cool but redundant. I think it's much simpler to use the original waiter (above) to wait for any `InvalidPiggyback` event to pop up and then assert that it has been fully handled (integrates fully) inside the subsequent assertion [here](https://github.com/omisego/elixir-omg/pull/422/files#diff-f77acdf5f259429a6dab36a42566b19eR117), by extending it with `outputs: [...], inputs: [...]`

Reply:
Tried removing it, turned out it helps with flakyness caused by async processing of IFE double-spending the piggybacked input. Fixed by waiting for NonCanonicalIFE instead and checking full list next. Added comment explaining why.